chore: prepare release 2.6.0 (#3302)
mobile-dev-inc/Maestro: CHANGELOG.md

- Improved rendering of JS errors
- Make `maestro hierarchy` command output only JSON
- Improve performance when cancelling a screenshot operation
- Remove unnecessary device interaction after running commands that won't affect the app (e.g. `assertVisible`)
- Fix `assertScreenshot` match percent so it aligns with the configured threshold
- Add `file` attribute to JUnit report
- Make JUnit XML `id` and `classname` attributes configurable
- Include cloud URL and app binary ID in the HTML report
- Stop assuming a lowercase `.yaml` extension when parsing flow names
- MCP `cheat_sheet` works without Maestro Cloud authentication
- MCP `take_screenshot` downscales screenshots to 2000px
- Throw the correct test error when a timeout value is invalid
- Fail fast with a helpful message when an env value is null
- Warn when the deprecated `--android-api-level` / `--ios-version` flags are used
- Fix `inputText` for native date inputs on web
- Swallow CDP `ServiceConfigurationError` in `WebDriver.open()`
- Fix `androidWebViewHierarchy: devtools` by preserving hostname in `DummyDns`
- Use emoji in Windows terminal output - consistent with other platforms

> - Bump Selenium to 4.43.0 and CDP bindings to v147
> - Additional CI for building Maestro on multiple OSes and adding console tests
> Thanks to @LVSant, @hSATAC, @mikamikaWolt, @dezsibalint, @sidferreira and @qwertey6 who contributed changes included in this release ❤️

Reviewer: ```suggestion
- Additional CI for building Maestro on multiple OSes and adding console tests

Thanks to @LVSant, @hSATAC, @mikamikaWolt, @dezsibalint, @sidferreira and @qwertey6 who contributed changes included in this release ❤️
```

## Unreleased

## 2.6.0

- Add Maestro Viewer (formerly MCP Visualizer) - a desktop visualizer for live hierarchy and commands
    - Pending-commands UX and panel redesign
    - Support physical Android devices

>     - Add `open_maestro_viewer` MCP tool

Reviewer: ```suggestion
- Add Maestro Viewer - a desktop visualizer for live hierarchy and commands
```

We don't need to include info about the WIP states

> - Remove bundled Maestro Studio from the CLI in favor of the Maestro desktop app

Reviewer: ```suggestion
- Remove bundled Maestro Studio from the CLI in favor of the Maestro Studio desktop app
```

- Remove the deprecated Rhino JS engine
- Enable parallel iOS simulator execution with per-device session tracking

> - Restore iOS simulator driver zips to their pre-#3138 state

Reviewer: ```suggestion
- Enable parallel iOS simulator execution with per-device session tracking
```

Remove WIP

- Enable gRPC channel-level retries on transient `UNAVAILABLE` in the Android driver
- Opt-in HTTP client retry on transient 5xx and `IOException`, enabled for XCTest

> - Route XCTest runner logs into per-run debug output

Reviewer: ```suggestion
- Route XCTest runner logs into per-run debug output
```

Remove reverted bits

> - Route XCTest runner logs into per-run debug output

Reviewer: ```suggestion
- Added iOS XCTest logs to debug output directory
```

> - Orchestra: structured parser errors with summary and detail

Reviewer: ```suggestion
- Improve YAML parsing error outputs
```

> - Use the literal `runScript` file as `sourceDescription` in Orchestra

Reviewer: ```suggestion
- Improved rendering of relative paths when using `runScript` commands
```

> - Attach YAML source provenance to `MaestroCommand`

Reviewer: ```suggestion
- Improved rendering of the source of failing commands
```

> - Convert `PolyglotException` to `JsEvaluationException` at the JS engine boundary

Reviewer: ```suggestion
- Improved rendering of JS errors
```

> - Make `hierarchy` command JSON-only

Reviewer: ```suggestion
- Make `maestro hierarchy` command output only JSON
```

> - Make `DeviceService.getDeviceScreen` suspend

Reviewer: ```suggestion
- Improve performance when cancelling a screenshot operation
```

> - Fix mutation setting of hierarchy-inert commands

Reviewer: ```suggestion
- Remove unnecessary device interaction after running commands that won't affect the app (e.g. `assertVisible`)
```

- Fix `assertScreenshot` match percent so it aligns with the configured threshold
- Add `file` attribute to JUnit report
- Make JUnit XML `id` and `classname` attributes configurable
- Include cloud URL and app binary ID in the HTML report
- Stop assuming a lowercase `.yaml` extension when parsing flow names
- MCP `cheat_sheet` works without Maestro Cloud authentication
- MCP `take_screenshot` downscales screenshots to 2000px
- Throw the correct test error when a timeout value is invalid
- Fail fast with a helpful message when an env value is null
- Warn when the deprecated `--android-api-level` / `--ios-version` flags are used
- Fix `inputText` for native date inputs on web
- Swallow CDP `ServiceConfigurationError` in `WebDriver.open()`

> - Preserve hostname in `DummyDns` so devtools augmentation works

Reviewer: ```suggestion
- Fix `androidWebViewHierarchy: devtools` by preserving hostname in `DummyDns`
```

> - Use emoji on Windows

Reviewer: ```suggestion
- Use emoji in Windows terminal output - consistent with other platforms
```

> - Bump Selenium to 4.43.0 and CDP bindings to v145

Reviewer: ```suggestion
- Bump Selenium to 4.43.0 and CDP bindings to v147
```

> - Add CI job to test building on all three OSes and verify version output

Reviewer: ```suggestion
- Additional CI for building Maestro on multiple OSes and adding console tests
```

- Add demo_app feature exposing the `assertScreenshot` threshold issue
- Update `.idea` for new Flutter additions
- Bump dependencies in GitHub Actions
- Add `bump-android-version` skill and `diagnose-maestro-failure` agent

> 

Reviewer: ```suggestion

```

Remove boring project CI things

> Thanks to @LVSant, @hSATAC, @mikamikaWolt, @dezsibalint and @qwertey6 who contributed changes included in this release ❤️

Reviewer: ```suggestion
Thanks to @LVSant, @hSATAC, @mikamikaWolt, @dezsibalint, @sidferreira and @qwertey6 who contributed changes included in this release ❤️
```

Add Sid, who did most of the legwork on the MCP image downsampling
